tests/lib3/test_constructor.py

```diff
@@ -222,6 +223,9 @@ def tzname(self, dt):
         def dst(self, dt):
             return datetime.timedelta(0)
 
+    class MyFullLoader(yaml.FullLoader):
+        state_blacklist_regexp = re.compile('(^mymethod$|^wrong_.*$)')
```

Comment:
This seems like it should be `blacklist_state_keys_regexp` to match `BLACKLIST_STATE_KEYS`.

I would prefer `STATE_KEYS_BLACKLIST`.

For the user blacklisting API I would prefer it not to an alternation regexp, but a list of regexps that you join together or loop over.

I think the user supplied list should be added onto the default list, and that a value of `None` resets the list. So by default the user is adding to the list (which may grow in the future) but to override completely, they start it with `None`.

So to be clear, change this line to:
```
state_keys_blacklist = ['^mymethod$', '^wrong_.*$']
```

Reply:
I was not able to come up with a solution using just simple fields, so I implemented `_get_state_keys_blacklist` and `_get_state_keys_blacklist_regexp`. Subclasses can define their own `_get_state_keys_blacklist` however they want (either by adding to the "super" blacklist or by just replacing it). Please tell me if it's ok, otherwise we can come up with something else.